dask/dataframe/io/parquet/core.py

```diff
@@ -579,6 +581,18 @@ def to_parquet(
         if append:
             raise ValueError("Cannot use both `overwrite=True` and `append=True`!")
         if fs.exists(path) and fs.isdir(path):
+            # Ref: https://github.com/dask/dask/issues/7466
+            # Check that we are not writing to the same location we are reading from
+            graph = df.__dask_graph__()
+            for layer_name, layer in graph.layers.items():
+                if layer_name.startswith("read-parquet-"):
+                    for k, v in layer.items():
+                        # check the arguments of the callable to extract the path
+                        if os.path.dirname(v[1]["piece"][0]) == path:
```

Comment:
Actually, this should probably not be `os.path`, because all paths ought to be posix-like.